docs/zero_to_hero_guide/00_Inference101.ipynb

```diff
@@ -0,0 +1,419 @@
+    "await chat_loop()\n",
+    "# To run it in a python file, use this line instead\n",
+    "# asyncio.run(chat_loop())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "8cf0d555",
+   "metadata": {},
+   "source": [
+    "## Conversation History\n",
+    "\n",
+    "Maintaining a conversation history allows the model to retain context from previous interactions. Use a list to accumulate messages, enabling continuity throughout the chat session."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "9496f75c",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User>  what is 1+1\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[36m> Response: 1 + 1 = 2\u001b[0m\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User>  what is llama + alpaca\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[36m> Response: That's a creative and imaginative question. However, since llamas and alpacas are animals, not numbers, we can't perform a mathematical operation on them.\n",
+      "\n",
+      "But if we were to interpret this as a creative or humorous question, we could say that the result of \"llama + alpaca\" is a fun and fuzzy bundle of South American camelid cuteness!\u001b[0m\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User>  what was the first question\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[36m> Response: The first question was \"what is 1+1\"\u001b[0m\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "User>  exit\n"
+     ]
+    },
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[33mEnding conversation. Goodbye!\u001b[0m\n"
+     ]
+    }
+   ],
+   "source": [
+    "async def chat_loop():\n",
+    "    conversation_history = []\n",
+    "    while True:\n",
+    "        user_input = input('User> ')\n",
+    "        if user_input.lower() in ['exit', 'quit', 'bye']:\n",
+    "            cprint('Ending conversation. Goodbye!', 'yellow')\n",
+    "            break\n",
+    "\n",
+    "        user_message = UserMessage(content=user_input, role='user')\n",
+    "        conversation_history.append(user_message)\n",
+    "\n",
+    "        response = client.inference.chat_completion(\n",
+    "            messages=conversation_history,\n",
+    "            model='Llama3.1-8B-Instruct',\n",
+    "        )\n",
+    "        cprint(f'> Response: {response.completion_message.content}', 'cyan')\n",
+    "\n",
+    "        assistant_message = UserMessage(content=response.completion_message.content, role='user')\n",
```

Comment:
let's not use these `UserMessage`, `SystemMessage`, classes please. just use regular old dicts with `role=user, content=...`

Reply:
thanks, fixed this and the other occurrence on the other notebooks